snow/networking/handler/message_queue.go

```diff
@@ -176,16 +171,17 @@ func (m *messageQueue) Len() int {
 	m.cond.L.Lock()
 	defer m.cond.L.Unlock()
 
-	return len(m.msgAndCtxs)
+	return m.msgAndCtxs.Len()
 }
 
 func (m *messageQueue) Shutdown() {
 	m.cond.L.Lock()
 	defer m.cond.L.Unlock()
 
 	// Remove all the current messages from the queue
-	for _, msg := range m.msgAndCtxs {
-		msg.msg.OnFinishedHandling()
+	for m.msgAndCtxs.Len() > 0 {
+		msgAndCtx, _ := m.msgAndCtxs.PopLeft()
+		msgAndCtx.msg.OnFinishedHandling()
 	}
 	m.msgAndCtxs = nil
```

Comment:
I think this changes the invariants such that `.Len()` and `.Shutdown()` after calling `.Shutdown()` will cause a panic.

Would it make more sense to just not `nil` out `msgAndCtxs`?

Reply:
Removed `m.msgAndCtxs = nil`